libfreerdp/core/client.c

```diff
@@ -246,7 +252,10 @@ int freerdp_channels_post_connect(rdpChannels* channels, freerdp* instance)
 
 			pChannelOpenData = &channels->openDataList[index];
 
-			pChannelClientData->pChannelInitEventProc(pChannelClientData->pInitHandle, CHANNEL_EVENT_CONNECTED, hostname, hostnameLength);
+			error = pChannelClientData->pChannelInitEventProc(pChannelClientData->pInitHandle,
+					CHANNEL_EVENT_CONNECTED, hostname, hostnameLength);
+			if (error != CHANNEL_RC_OK)
+				goto fail;
 
 			name = (char*) malloc(9);
 			if (!name)
```

Comment:
If name is NULL we return a signed value although the function's return type is UINT